release/update_and_create_archives.sh

```diff
@@ -83,9 +81,10 @@ if [[ -n "${GCR_TEST_PATH}" && -n "${GCS_TEST_PATH}" ]]; then
 
   DOCKER_HUB_TAG="gcr.io/${GCR_TEST_PATH},${VER_STRING}"
   COMMON_URL="https://storage.googleapis.com/${GCS_TEST_PATH}"
-  ISTIOCTL_URL="${COMMON_URL}/istioctl"
+  ISTIOCTL_URL="${COMMON_URL}/istioctl-stage"
```

Comment:
what's the -stage for ?

Reply:
In my earlier drafts I called it "istioctl-test" (to reflect it was used with the "testonly" tars) but had a change of heart to "istioctl-stage" (to reflect it's used with the staged location of the build).  I'm not hard-set on either name, or another name, and it may not matter if there's other changes later to reduce/consolidate tars or build storage locations to address some of your concerns.

The build has 3 basic stages: 1) generation of artifacts, 2) creation of tars, and 3) upload of artifacts/tars to gcs/gcr.  Step \#2 creates one set of tars that point to docker.io, and a second "testonly" set of tars that point to gcr.  Consequently \#1 now generates one set of istioctl that points to docker.io (which is stored under the usual istioctl/ ) and a second set of istioctl that point to to gcr (which is stored under the new istioctl-stage/ ).

If the tars merely contained a copy of istioctl then "istioctl-stage/" wouldn't need to live past step \#2.  However, the tars also contain a istio.VERSION file with a URL that points to the directory of the istioctl files.  Consequently for the "testonly" tars I need to now set ISTIOCTL_URL to point to the "istioctl-stage/" version of the files (which is the line you're asking about), and step \#3 now needs to upload "istioctl-stage" to GCS (I didn't have to make an explicit change for \#3 since it's handled for free by the existing "cp -r *").

There's now a slight bit of untidiness in how the release process (RP) runs the build pipeline (actually it was always there but is more explicit/obvious now).  The RP tells the build to store to a "daily-pipeline" location for GCS, and "istio-release" for GCR.  The RP then runs some tests and if they pass it copies the GCS artifacts to another "istio-prerelease" location in GCS (the GCR is left alone).  An untidy aspect is that the istioctl URL I mentioned will point to the "daily-pipeline location" rather than "istio-prerelease".  This isn't ideal (and I think people might've been under the impression that the GCS path wasn't stored anywhere in the build artifacts), though I don't think it should break anything either since I think both buckets are publically accessible.   cc @hklai

Reply:
Ic, nice, thanks for the explanation!